Formula/nest.rb

```diff
@@ -16,35 +16,43 @@ class Nest < Formula
   depends_on "cmake" => :build
   depends_on "cython" => :build
   depends_on "gsl"
-  depends_on "libomp"
   depends_on "libtool"
   depends_on "numpy"
   depends_on "python@3.9"
   depends_on "readline"
 
-  def install
-    args = ["-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}"]
+  uses_from_macos "ncurses"
 
-    libomp = Formula["libomp"]
-    args << "-Dwith-openmp=ON"
-    args << "-Dwith-libraries=#{libomp.opt_lib}/libomp.dylib"
-    args << "-DOpenMP_C_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
-    args << "-DOpenMP_C_LIB_NAMES=omp"
-    args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
-    args << "-DOpenMP_CXX_LIB_NAMES=omp"
-    args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
+  on_macos do
+    depends_on "libomp"
+  end
 
-    mkdir "build" do
-      system "cmake", "..", *args
-      system "make"
-      system "make", "install"
+  def install
+    args = %W[
+      -DCMAKE_INSTALL_PREFIX:PATH=#{prefix}
+      -Dwith-openmp=ON
+    ]
+    if OS.mac?
+      libomp = Formula["libomp"]
+      args << "-Dwith-libraries=#{libomp.opt_lib}/libomp.dylib"
+      args << "-DOpenMP_C_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
+
+      # Help FindReadline find system ncurses library
+      lib_suffix = (MacOS.version >= :big_sur) ? "tbd" : "dylib"
+      args << "-DNCURSES_LIBRARY=#{MacOS.sdk_path_if_needed}/usr/lib/libncurses.#{lib_suffix}"
     end
 
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
     # Replace internally accessible gcc with externally accessible version
     # in nest-config if required
-    inreplace bin/"nest-config",
-        %r{#{HOMEBREW_REPOSITORY}/Library/Homebrew/shims.*/super}o,
-        "#{HOMEBREW_PREFIX}/bin"
+    inreplace bin/"nest-config", Superenv.shims_path/ENV.cxx, ENV.cxx
```

Comment:
Previously was setting `$HOMEBREW_PREFIX/bin/clang++`, which wouldn't exist unless user installed `llvm` and then manually linked it as keg_only.